### PR TITLE
Fix styling in primary and secondary tagging

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses #4119

## Changes

* Fix styling in primary and secondary tagging

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/views/AnalyticalFramework/FrameworkForm/PrimaryTagging/SectionsEditor/index.tsx [hunk 1]
```diff
@@ -276,11 +295,14 @@ function SectionsEditor(props: Props) {
...
         index,
+        onExpansionChange: handleExpansionChange,
+        expanded: expandedSectionId === section.clientId,
     }), [
         onSectionsChange,
         focusedSection,
         onSectionsRemove,
         arrayError,
+        expandedSectionId,
     ]);
```
Comment (samshara) [inline app/views/AnalyticalFramework/FrameworkForm/PrimaryTagging/SectionsEditor/index.tsx]: add ```handleExpansionChange``` to useCallback dependecy list

File: app/views/AnalyticalFramework/FrameworkForm/components/WidgetEditor/MultiSelectWidgetForm/index.tsx [hunk 2]
```diff
@@ -268,19 +282,23 @@ function DataInput<K extends string>(props: DataInputProps<K>) {
...
         autoFocus: newlyCreatedOptionIdRef.current === option.clientId,
         index,
+        onExpansionChange: handleExpansionChange,
+        expanded: expandedOptionId === option.clientId,
     }), [
         onOptionsChange,
         onOptionsRemove,
         arrayError,
+        expandedOptionId,
     ]);
```
Comment (samshara) [inline app/views/AnalyticalFramework/FrameworkForm/components/WidgetEditor/MultiSelectWidgetForm/index.tsx]: add ```handleExpansionChange``` to useCallback dependency list

File: app/views/AnalyticalFramework/FrameworkForm/components/WidgetEditor/ScaleWidgetForm/index.tsx [hunk 3]
```diff
@@ -308,12 +325,15 @@ function DataInput<K extends string>(props: DataInputProps<K>) {
...
         autoFocus: newlyCreatedOptionIdRef.current === option.clientId,
+        onExpansionChange: handleExpansionChange,
+        expanded: expandedScaleId === option.clientId,
     }), [
         onOptionsChange,
         handleOptionRemove,
         handleDefaultValueChange,
         value?.defaultValue,
         arrayError,
+        expandedScaleId,
```
Comment (samshara) [inline app/views/AnalyticalFramework/FrameworkForm/components/WidgetEditor/ScaleWidgetForm/index.tsx:336]: add ```handleExpansionChange``` to dependency list

File: app/views/AnalyticalFramework/FrameworkForm/components/WidgetEditor/SingleSelectWidgetForm/index.tsx [hunk 4]
```diff
@@ -269,10 +283,13 @@ function DataInput<K extends string>(props: DataInputProps<K>) {
         value: option,
         autoFocus: newlyCreatedOptionIdRef.current === option.clientId,
         index,
+        onExpansionChange: handleExpansionChange,
+        expanded: expandedOptionId === option.clientId,
     }), [
         onOptionsChange,
         onOptionsRemove,
         arrayError,
+        expandedOptionId,
```
Comment (samshara) [inline app/views/AnalyticalFramework/FrameworkForm/components/WidgetEditor/SingleSelectWidgetForm/index.tsx:292]: same as above

File: app/views/AnalyticalFramework/FrameworkForm/PrimaryTagging/SectionsEditor/styles.css [hunk 5]
```diff
@@ -1,18 +1,55 @@
...
+                            padding: calc(var(--spacing) / 2) 0;
+                        }
+                    }
+                    .text-input {
+                        margin: var(--dui-spacing-small) 0;
+                    }
+                }
+            }
+        }
     }
```
Comment (tnagorra) [inline app/views/AnalyticalFramework/FrameworkForm/PrimaryTagging/SectionsEditor/styles.css:36]: Should we need to manually modify padding for these many components? @frozenhelium

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Comment by tnagorra** [comment]
Please fix these warning
image

File: app/views/AnalyticalFramework/FrameworkForm/PrimaryTagging/SectionsEditor/styles.css (hunk 5 above)
Comment (subinasr) [reply app/views/AnalyticalFramework/FrameworkForm/PrimaryTagging/SectionsEditor/styles.css:36]: Yes, it's needed to achieve the look we're going for!

**Pull request merged**